sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/internal/SubscriptionPartitionProcessorImpl.java

```diff
@@ -114,26 +97,50 @@ public void failed(State from, Throwable failure) {
     completionFuture.setException(ExtractStatus.toCanonical(failure));
   }
 
+  @SuppressWarnings("argument.type.incompatible")
+  private void start() throws ApiException {
+    this.subscriber.addListener(this, SystemExecutors.getFuturesExecutor());
+    this.subscriber.startAsync();
+    this.subscriber.awaitRunning();
+    try {
+      this.subscriber.allowFlow(
```

Comment:
It looks like there is a way to provide flow control without buffering. Would one of the BlockingSubscriber implementations help simplify the contract here?

Reply:
No, but I've updated this.